common/namespace/namespace.go

```diff
@@ -89,6 +89,24 @@ const (
 	EmptyID   ID   = ""
 )
 
+var (
+	errNamespaceTooLong = serviceerror.NewInvalidArgument("Namespace length exceeds limit.")
+)
+
+// Return a valid Name if and only if the provided string represents a valid
+// namespace name.
+// Note that an empty string is a valid namespace name as the name itself is optional in places
+// TODO: empty names should be invalid and return a serviceerror
+func ParseName(name string, maxLength int) (Name, error) {
```

Comment:
I'm a fan of the "parse, don't validate" approach to code but I can bend on this if we care